src/Digdir.Domain.Dialogporten.Domain/Parties/Abstractions/PartyIdentifier.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+namespace Digdir.Domain.Dialogporten.Domain.Parties.Abstractions;
+
+public static class PartyIdentifier
+{
+    private delegate bool TryParseDelegate(ReadOnlySpan<char> value, [NotNullWhen(true)] out IPartyIdentifier? identifier);
+    private static readonly Dictionary<string, TryParseDelegate> TryParseByPrefix = CreateTryParseByPrefix();
+    private const string Separator = "::";
+
+    public static bool TryParse(ReadOnlySpan<char> value, [NotNullWhen(true)] out IPartyIdentifier? identifier)
+    {
+        identifier = null;
+        var separatorIndex = value.IndexOf(Separator);
+        if (separatorIndex == -1)
+        {
+            return false;
+        }
+
+        var prefix = value[..(separatorIndex + Separator.Length)].ToString();
+        return TryParseByPrefix.TryGetValue(prefix, out var tryParse)
+            && tryParse(value, out identifier);
+    }
+
+    internal static ReadOnlySpan<char> GetIdPart(ReadOnlySpan<char> value)
+    {
+        var separatorIndex = value.IndexOf(Separator);
+        return separatorIndex == -1
+            ? value
+            : value[(separatorIndex + Separator.Length)..];
+    }
+
+    private static Dictionary<string, TryParseDelegate> CreateTryParseByPrefix()
+    {
+        var partyIdentifiers = typeof(IPartyIdentifier)
+            .Assembly
+            .GetTypes()
+            .Where(type => type.IsClass && !type.IsAbstract && type.IsAssignableTo(typeof(IPartyIdentifier)))
+            .Select(partyIdentifierType => new
+            {
+                Type = partyIdentifierType,
+                Prefix = (string)partyIdentifierType
+                    .GetProperty(nameof(IPartyIdentifier.Prefix), BindingFlags.Static | BindingFlags.Public)!
+                    .GetValue(null)!,
+                TryParse = partyIdentifierType
+                    .GetMethod(nameof(IPartyIdentifier.TryParse), new[] { typeof(ReadOnlySpan<char>), typeof(IPartyIdentifier).MakeByRefType() })!
+                    .CreateDelegate<TryParseDelegate>()
+            })
+            .ToList();
+
+        var nullOrWhitespacePrefix = partyIdentifiers
```

Comment:
Kunne kanskje vurdert å trekke ut de tre formatsjekkene i en egen, eller 3 egne, separata metoder for å senke kompleksiteten noe.

Eks:
AssertPrefixNotNullOrWhitespace(partyIdentifiers);
AssertPrefixEndsWithSeparator(partyIdentifiers);
AssertNoIdenticalPrefixes(partyIdentifiers);

Reply:
Liker det, will do! 🙏